src/equiv.js

```diff
@@ -239,37 +217,23 @@ QUnit.equiv = (function() {
 	};
 
 	function typeEquiv( a, b ) {
-		var prop = QUnit.objectType( a );
-		return callbacks[ prop ]( b, a );
+		var type = QUnit.objectType( a );
+		return QUnit.objectType( b ) === type && callbacks[ type ]( b, a );
```

Comment:
Until checking this change above, it is hard to see why the `"nan"` check is returning true only.